wled00/pin_manager.cpp

```diff
@@ -267,6 +267,14 @@ bool PinManagerClass::isPinOk(byte gpio, bool output) const
   return false;
 }
 
+bool PinManagerClass::isReadOnlyPin(byte gpio)
+{
+#ifdef ARDUINO_ARCH_ESP32
+  if (gpio < WLED_NUM_PINS) return digitalPinCanOutput(gpio);
```

Comment:
This should be !digitalPinCanOutput().

Unfortunately even with that fixed it still doesn't work because this function also returns FALSE on other pins than R/O (for example 28 to 31 on ESP32). 

While as far as the finalizeInit() code this doesn't matter I think we should not return TRUE for pins that are not read/only on a function called isReadOnlyPin(). This could lead to bugs in the future (e.g. usermod wanting to use this function to check if a pin can be used to read data).

There are two ways I can think of to get around this problem:
1. Hardcode the read only pins (which I know you're not a fan of).
2. Remove the orange coloring in the GUI for read/only pins. We don't need to check if a pin is r/o in finalizeInit() and if the r/o pins are red (since they fail the digitalPinCanOutput() check) in the GUI instead of orange I don't think it's a big deal. As far as I can tell there's no other place in the codebase that does a r/o check, but please correct me if I'm wrong.

Reply:
> Unfortunately even with that fixed it still doesn't work because this function also returns FALSE on other pins than R/O (for example 28 to 31 on ESP32).

This is used in pin manager since ever:
https://github.com/Aircoookie/WLED/blob/df24fd7bf289dd3fe86fdb8776b0d25bbf3d9b30/wled00/pin_manager.cpp#L259

If you do not trust HAL (you may have reasons) then use `isPinOk(gpio) != isPinOk(gpio,false)` to determine RO pins. This way there will only be one place to configure GPIOs and that is `isPinOk()`.

Reply:
> > Unfortunately even with that fixed it still doesn't work because this function also returns FALSE on other pins than R/O (for example 28 to 31 on ESP32).
> 
> This is used in pin manager since ever:
> 
> https://github.com/Aircoookie/WLED/blob/df24fd7bf289dd3fe86fdb8776b0d25bbf3d9b30/wled00/pin_manager.cpp#L259
> 
> If you do not trust HAL (you may have reasons) then use `isPinOk(gpio) != isPinOk(gpio,false)` to determine RO pins. This way there will only be one place to configure GPIOs and that is `isPinOk()`.

Sorry, I had already implemented the logic fix before reading your message. Both options work the same in my testing so your call on which you prefer:

pinManager.isPinOk(i) != pinManager.isPinOk(i,false)

digitalPinIsValid(gpio) && !digitalPinCanOutput(gpio)